integration/token/fungible/views/issue.go

```diff
@@ -65,14 +67,25 @@ func (p *IssueCashView) Call(context view.Context) (interface{}, error) {
 	}
 
 	// At this point, the issuer is ready to prepare the token transaction.
-	// The issuer creates an anonymous transaction (this means that the resulting Fabric transaction will be signed using idemix, for example),
-	// and specify the auditor that must be contacted to approve the operation
-	tx, err := ttx.NewAnonymousTransaction(
-		context,
-		ttx.WithAuditor(
-			view2.GetIdentityProvider(context).Identity("auditor"), // Retrieve the auditor's FSC node identity
-		),
-	)
+	// The issuer creates a new token transaction and specify the auditor that must be contacted to approve the operation.
+	var tx *ttx.Transaction
+	if p.Anonymous {
+		tx, err = ttx.NewAnonymousTransaction(
+			context,
+			ttx.WithAuditor(
+				view2.GetIdentityProvider(context).Identity("auditor"), // Retrieve the auditor's FSC node identity
+			),
```

Comment:
could you store the `WithAuditor` option in a var before and use it in the if/else clause?

Also, there might be an issue if `NewTransaction` or `NewAnonymousTransaction` returns an error and `tx` becomes `nil`. I suggest that you move the err check right after the if/else before setting metadata.

Reply:
Actually, I am wondering why we have `NewAnonymousTransaction` and `NewTransaction`. All the parameters of these functions could be provided via options, can't we?

Reply:
true, but with a clean different API, it is difficult to get it wrong 😅

Reply:
I'll move the error check 👍